### PR TITLE
[Intl] do not access typed property before initialization

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix https://github.com/symfony/symfony/pull/58600#discussion_r1806632284
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Intl/Transliterator/EmojiTransliterator.php [hunk 1]
```diff
@@ -72,12 +72,12 @@ public function createInverse(): self
 
         public function getErrorCode(): int|false
         {
-            return $this->transliterator?->getErrorCode() ?? 0;
+            return isset($this->transliterator) ? $this->transliterator->getErrorCode() : 0;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Intl/Transliterator/EmojiTransliterator.php:75]: Is this really needed? Do the tests fail without the patch? Doesn't the `??` already guard against these?

Comment (xabbuh) [reply src/Symfony/Component/Intl/Transliterator/EmojiTransliterator.php:75]: The null-coalesce operator doesn't help here when calling a method (that's why this went unnoticed in the linked discussion). The tests are failing without these changes (see also https://3v4l.org/HBV0b).

File: src/Symfony/Component/Intl/Transliterator/EmojiTransliterator.php [hunk 2]
```diff
@@ -72,12 +72,12 @@ public function createInverse(): self
...
         }
 
         public function getErrorMessage(): string|false
         {
-            return $this->transliterator?->getErrorMessage() ?? false;
+            return isset($this->transliterator) ? $this->transliterator?->getErrorMessage() : false;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Intl/Transliterator/EmojiTransliterator.php]:
```suggestion
            return isset($this->transliterator) ? $this->transliterator->getErrorMessage() : false;
```

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Intl/Transliterator/EmojiTransliterator.php (hunk 2 above)
Comment (xabbuh) [reply src/Symfony/Component/Intl/Transliterator/EmojiTransliterator.php]: oops 😆

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**